subsys/bluetooth/host/gatt.c

```diff
@@ -621,11 +622,38 @@ struct notify_data {
 	struct bt_gatt_indicate_params *params;
 };
 
-static int gatt_notify(struct bt_conn *conn, u16_t handle, const void *data,
+void bt_gatt_notification_cb_register(struct bt_gatt_notify_cb *cb)
+{
+	cb->_next = callback_list;
+	callback_list = cb;
+}
+
+static void notify_notification_tx_complete(struct bt_conn *conn,
+					    const struct bt_gatt_attr *attr)
+{
+	struct bt_gatt_notify_cb *cb;
+
+	for (cb = callback_list; cb; cb = cb->_next) {
+		if (cb->notify_complete) {
```

Comment:
Instead of checking null pointer in loop here, please check for null pointers when registering the callback.